GoogleForCreators/web-stories-wp
# Avoid duplicate poster generation

## Summary

Refreshes media modal to ensure it has all the data about poster images etc.

## Relevant Technical Choices

* Refreshes media modal to ensure it has all the data about poster images etc.  
  See https://github.com/google/web-stories-wp/issues/5675#issuecomment-745591608 for explainer
* When inserting a video from the WordPress media modal, do not try to generate a poster image if the video already has a poster.

## To-do



## User-facing changes

* Media modal refreshes when there's newly uploaded media

## Testing Instructions

1. In the WordPress media library (Media -> Add New), upload a video
2. Open dev tools and monitor network tab
2. Create a new story
3. See that video is available in media pane.
4. Optional: Verify that in the network tab you see some XHR requests to `web-stories/v1/media/..` -> a poster is being generated!
5. Click on "Upload" to open media dialog, switch to "Gallery"
6. See that the new video now has a poster, and both the video and the poster are in the list of media items that can be selected (so video + poster = 2 items)
7. Create a new story
8. Click on "Upload", switch to "Gallery"
9. Insert the "video" into page
10. Optional: See that there are _no_ new XHR requests to `web-stories/v1/media/..` -> no extra poster is being generated
11. Click on "Upload" again to open media dialog, switch to "Gallery"
12. See that the video still has a poster, and _no new poster_ has been added. So you still see the video + the poster in the list of selectable items, but not another new one (so video + poster = 2 items)

**Note:** In the WordPress media modal and the WordPress media library, you will see both the video and its poster image, whereas in the story editor the poster is not visible; only the video. This is currently expected.

---



Fixes #5675
Fixes #5541
